addons/mail/static/src/core/web/message_patch.js

```diff
@@ -181,39 +181,35 @@ patch(Message.prototype, {
             case "char":
             case "many2one":
             case "selection":
-                return formatChar(trackingValue.value);
+                return formatChar(trackingValue);
             case "date": {
-                const value = trackingValue.value
-                    ? deserializeDate(trackingValue.value)
-                    : trackingValue.value;
+                const value = trackingValue ? deserializeDate(trackingValue) : trackingValue;
                 return formatDate(value);
             }
             case "datetime": {
-                const value = trackingValue.value
-                    ? deserializeDateTime(trackingValue.value)
-                    : trackingValue.value;
+                const value = trackingValue ? deserializeDateTime(trackingValue) : trackingValue;
                 return formatDateTime(value);
             }
             case "float":
-                return formatFloat(trackingValue.value);
+                return formatFloat(trackingValue);
             case "integer":
-                return formatInteger(trackingValue.value);
+                return formatInteger(trackingValue);
             case "text":
-                return formatText(trackingValue.value);
+                return formatText(trackingValue);
             case "monetary":
-                return formatMonetary(trackingValue.value, {
-                    currencyId: trackingValue.currencyId,
+                return formatMonetary(trackingValue, {
+                    currencyId: trackingFieldInfo.currencyId,
                 });
             default:
-                return trackingValue.value;
+                return trackingValue;
         }
     },
 
     /**
      * @returns {string}
      */
-    formatTrackingOrNone(trackingType, trackingValue) {
-        const formattedValue = this.formatTracking(trackingType, trackingValue);
+    formatTrackingOrNone(trackingFieldInfo, trackingValue) {
+        const formattedValue = this.formatTracking(trackingFieldInfo.fieldType, trackingValue);
```

Comment:
Seems weird, you assume in the `formatTracking` signature that you have an Object but here you give it the type directly. Which is it? (Maybe add an update to the JSDoc when you have made your choice)

Reply:
Up on this one.

Reply:
Ah yeah :thinking: 
Should just be fieldInfo indeed